multiversx/mx-chain-go
# Optimization: encode hash to hex only on log TRACE level set

## Reasoning behind the pull request

Reasonable optimization on trie updates. Improved update times by roughly 30%.

Before fix:

```
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkPatriciaMerkleTrie_Update-8   	      50	2007485284 ns/op
ok  	github.com/multiversx/mx-chain-go/trie	131.749s

BenchmarkPatriciaMerkleTrie_Update-8   	      50	1991358120 ns/op
ok  	github.com/multiversx/mx-chain-go/trie	132.290s
```

After fix:

```
BenchmarkPatriciaMerkleTrie_Update-8   	      50	1342203572 ns/op
ok  	github.com/multiversx/mx-chain-go/trie	95.049s

BenchmarkPatriciaMerkleTrie_Update-8   	      50	1381106022 ns/op
ok  	github.com/multiversx/mx-chain-go/trie	97.326s
```

## Proposed changes
- Added if statement before logging.

## Testing procedure
- `go test -run=^$ -bench ^BenchmarkPatriciaMerkleTrie_Update$ -benchtime=10x`
- Observe times with from the target branch and compare with the times from this branch.
- Standard testing

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
